lib/acts_as_taggable/acts_as_taggable.rb

```diff
@@ -114,7 +114,8 @@ def tag_names
     end
 
     def tag_names=(names)
-      send HelperMethods.scoped_association_assignment_name, names.select(&:present?).collect {|tag_name| self.class.create_tag(tag_name) }
+      tag_objects = names.select(&:present?).collect {|tag_name| self.class.create_tag(tag_name) }.uniq
```

Comment:
Why are we not calling `uniq` on the `names` before we call `collect { create_tag }`?

Reply:
That could make sense, too.  Is there a possible case where `Tag#sanitize` does some of its magic and gives us duplicates, e.g. `names = ["Tag ", " tag ", "   tAg"]` so we'd want to call `uniq` on the actual tag records?

Reply:
Ah yes good point. This is better than my suggestion then :).